src/test/java/io/vlingo/actors/ContentBasedRoutingStrategyTest.java

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+
+package io.vlingo.actors;
+
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import org.junit.Test;
+
+import io.vlingo.actors.testkit.TestUntil;
+
+/**
+ * ContentBasedRoutingStrategyTest
+ */
+public class ContentBasedRoutingStrategyTest {
+
+  @Test
+  public void testThatItRoutes() throws InterruptedException {
+    final World world = World.startWithDefaults("RandomRouterTest");
+    final int poolSize = 4;
+    final TestUntil until = TestUntil.happenings(40);
+    final OrderRouter orderRouter = world.actorFor(
+            Definition.has(OrderRouterActor.class, Definition.parameters(poolSize, until)),
+            OrderRouter.class);
+    
+    String[] customerIds = {"Customer1", "Customer2", "Customer3","Customer4"};
+    Random random = new Random();
+    
+    /* is this the right approach? */
+    while (until.remaining() > 0) {
```

Comment:
This is mostly unnecessary since you can send some count (40) messages and the `until.completes()` will block until the number of expected `until.happened()` occur.